### PR TITLE
Use the shared publish workflow

### PR DESCRIPTION
**Pull request by AndrewGable**
Related: https://github.com/Expensify/GitHub-Actions/pull/1
Related: https://github.com/Expensify/Expensify/issues/432173

### CODE REVIEW

**Comment by AndrewGable** [comment]
@johnmlee101 - We will have to add the `OS_BOTIFY_COMMIT_TOKEN` to the org (or this repo)

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -27,71 +24,10 @@ jobs:
...
-
-      - name: Set up git for OSBotify
-        run: |
-          git config --global user.signingkey AEE1036472A782AB
-          git config --global commit.gpgsign true
-          git config --global user.name OSBotify
-          git config --global user.email infra+osbotify@expensify.com
-
-      - uses: actions/setup-node@v3
+      - uses: Expensify/GitHub-Actions/.github/workflows/npmPublish.yml@main
```
Comment (johnmlee101) [inline .github/workflows/publish.yml:27]: FYI I wonder if we should create versions for this whenever we make code changes in this repo.

**Comment by johnmlee101** [comment]
`OS_BOTIFY_COMMIT_TOKEN` yeah we can add it to the org, that's fine

**Comment by johnmlee101** [comment]
well org with whitelisted repos

File: .github/workflows/publish.yml (hunk 1 above)
Comment (johnmlee101) [reply .github/workflows/publish.yml:27]: also for testing you can pin it to your branch I think?

Comment (AndrewGable) [reply .github/workflows/publish.yml:27]: Once I confirm it works, we can tag that commit as 1.0 and update all `main` references to that 👍

**Comment by AndrewGable** [comment]
Must be holding it wrong: https://github.com/Expensify/eslint-config-expensify/actions/runs/11278808120

**Review by johnmlee101 — APPROVED** [review]

**Pull request merged**